src/Sulu/Bundle/Sales/CoreBundle/Item/ItemManager.php

```diff
@@ -644,6 +644,15 @@ protected function setItemByProductData($productData, ApiItemInterface $item, $l
         $item->setIsRecurringPrice($product->isRecurringPrice());
 
         $taxClass = $product->getTaxClass();
+
+        /*
+         * If the product doesn't have a taxClass but has a parent, we try to get the taxClass from the parent.
+         * This is used for product variants, which get the taxClass from the original product.
+         */
+        if (!$taxClass && $product->getParent()) {
```

Comment:
i'd move that as elseif to the if clause below ;)

Reply:
I don't think that would be an improvement, because taxclass could still be null (if the parent doesn't have one as well) 
See:

```
if ($taxClass) {
            $tax = $this->retrieveTaxForClass($taxClass);
            $item->setTax($tax);
        } elseif ($product->getParent()) {
            $taxClass = $product->getParent()->getTaxClass();
            if ($taxClass) {
                $tax = $this->retrieveTaxForClass($taxClass);
                $item->setTax($tax);
            }
        } else {
            $item->setTax(0);
        }
```

Reply:
yeah, didn't see that it's causing the logic to be duplicated. you are allowed to leave it as you suggested 😎